client/js/helpers/contextMenu.js

```diff
@@ -183,6 +183,43 @@ export function generateChannelContextMenu($root, channel, network) {
 	return items;
 }
 
+export function generateInlineChannelContextMenu($root, chan, network) {
+	const join = () => {
+		const channel = network.channels.find((c) => c.name === chan);
+
+		if (channel) {
+			$root.switchToChannel(channel);
+		}
+
+		socket.emit("input", {
+			target: $root.$store.state.activeChannel.channel.id,
+			text: "/join " + chan,
+		});
+	};
+
+	const items = [];
+
+	const channel = network.channels.find((c) => c.name === chan);
+
+	if (channel) {
```

Comment:
I would rather `return [{ }]` twice than have the single return and the `items = []` declaration.